lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -94,13 +98,37 @@ export default function FilePickerApp({ onSubmit }) {
   // Submit the form after a selection is made via one of the available
   // methods.
   useEffect(() => {
+    async function createAssignment() {
+      const data = { ...createAssignmentAPI.data };
+      data.content = content;
+      data.groupset = groupConfig.groupSet;
+      const assignment = await apiCall({
+        authToken,
+        path: createAssignmentAPI.path,
+        data: data,
+      });
+      setExtLTIAssignmentId(assignment.ext_lti_assignment_id);
+    }
+
+    if (content && createAssignmentAPI && !extLTIAssignmentId) {
+      createAssignment();
+      return;
+    }
+
     if (shouldSubmit) {
       // Submit form using a hidden button rather than calling `form.submit()`
       // to facilitate observing the submission in tests and suppressing the
       // actual submit.
       submitButton.current.click();
     }
-  }, [shouldSubmit]);
+  }, [
+    shouldSubmit,
+    extLTIAssignmentId,
+    authToken,
+    content,
+    createAssignmentAPI,
+    groupConfig.groupSet,
```

Comment:
This seems like a long list but eslit suggested that.

Reply:
The idea is that effects are informed of their dependencies so they re-run when any of them change. In this case, can any of the values change after all the requirements are met to trigger the `createAssignment` call (the `content && createAssignmentAPI && !extLTIAssignmentId` check above)?

Reply:
Ah - I see that the `!extLTIAssignmentId` check at the end ensures that we only call this once.